great_expectations/datasource/batch_kwargs_generator/query_batch_kwargs_generator.py

```diff
@@ -37,15 +64,19 @@ def __init__(
         super(QueryBatchKwargsGenerator, self).__init__(
             name=name, datasource=datasource
         )
-        root_directory = None
+        if (
```

Comment:
Nonblocking: This kind of guessing logic still makes me nervous. 

Can we articulate when it's okay for a thing to guess, and when non-null typed (or duck-typed) objects must be passed?

Reply:
I definitely think it's a good idea to have a "coherent" picture of when we have things guess.

Rather than a "guess" in this case, I think this is just a default value. The if statements could be replaced with try/catch in this case.